app/components/reports/reconciliation.tsx

```diff
@@ -0,0 +1,349 @@
+
+  if (loading) return <Text>{LL.common.soon()}</Text>
+  if (error) return <Text>{LL.common.error()}</Text>
+
+  const filteredTransactionsByDirection = filterTransactionsByDirection(
+    filterTransactionsByDate(
+      data?.me?.defaultAccount?.transactions?.edges?.map((edge) => edge.node) || [],
+      from,
+      to,
+    ),
+    selectedDirection,
+  )
+
+  const orderedTransactions = orderAndConvertTransactionsByDate(
+    filteredTransactionsByDirection,
+    convertMoneyAmount,
+  )
+
+  return (
+    <Screen>
+      <View style={styles.container}>
+        <DateRangeDisplay from={from} to={to} />
+
+        <Text style={styles.totalText}>
+          {LL.reports.total()}:{"\nUSD"} {balance}{" "}
+          {balanceInDisplayCurrency !== balance && (
+            <Text>{`( ~${balanceInDisplayCurrency} )`}</Text>
+          )}
+        </Text>
+
+        <View style={styles.filterContainer}>
+          <Button
+            title="All"
+            type={selectedDirection === null ? "solid" : "outline"}
+            onPress={() => setSelectedDirection(null)}
+          />
+          <Button
+            title="Received"
+            type={selectedDirection === TxDirection.Receive ? "solid" : "outline"}
+            onPress={() => setSelectedDirection(TxDirection.Receive)}
+          />
+          <Button
+            title="Sent"
+            type={selectedDirection === TxDirection.Send ? "solid" : "outline"}
+            onPress={() => setSelectedDirection(TxDirection.Send)}
+          />
+        </View>
+        <ScrollView style={styles.scrollView}>
+          {orderedTransactions.map((tx) => (
+            <View key={tx.id} style={styles.transactionRow}>
+              <View style={styles.transactionDetailsRow}>
+                <Text style={styles.dateText}>{tx.displayDate}</Text>
+                <Text style={styles.txDirectionText}>{tx.direction}</Text>
+                <Text style={styles.amountText}>{tx.settlementDisplayAmount}</Text>
+              </View>
+            </View>
+          ))}
+        </ScrollView>
+        <Button
+          style={styles.button}
+          onPress={() =>
+            exportTransactionsToHTML({
+              transactions: orderedTransactions,
+              from,
+              to,
+              totalAmount: balance,
+              balanceInDisplayCurrency,
+              currencySymbol: "USD",
+            })
+          }
+        >
+          Export as HTML
+        </Button>
+        <Button
+          style={styles.button}
+          onPress={() =>
+            exportTransactionsToPDF({
+              transactions: orderedTransactions,
+              from,
+              to,
+              totalAmount: balance,
+              balanceInDisplayCurrency,
+              currencySymbol: "USD",
+            })
+          }
+        >
+          Export as PDF
+        </Button>
+      </View>
+    </Screen>
+  )
+}
+
+const filterTransactionsByDate = (
```

Comment:
Make util file in utils folder for filterTransactionsByDate, filterTransactionsByDirection, calculateTotalAmount, convertToDisplayCurrency and formatDate functions. Import in this file and use it to keep the screen file clean.